src/main/java/com/liveramp/daemon_lib/Daemon.java

```diff
@@ -90,6 +92,22 @@ public Daemon(String identifier, JobletExecutor<T> executor, JobletConfigProduce
     this.notifier = notifier;
     this.options = options;
     this.lock = lock;
+    this.wakeUpCallback = new DaemonCallback.None();
+
+    this.running = false;
+  }
+
+  public Daemon(String identifier, JobletExecutor<T> executor, JobletConfigProducer<T> configProducer, JobletCallback<? super T> preExecutionCallback, DaemonLock lock, DaemonNotifier notifier, Options options, ExecutionCondition executionCondition, ConfigBasedExecutionCondition<T> configBasedExecutionCondition, DaemonCallback wakeUpCallback) {
+    this.preExecutionCallback = preExecutionCallback;
+    this.executionCondition = executionCondition;
+    this.configBasedExecutionCondition = configBasedExecutionCondition;
+    this.identifier = clean(identifier);
+    this.configProducer = configProducer;
+    this.executor = executor;
+    this.notifier = notifier;
+    this.options = options;
+    this.lock = lock;
+    this.wakeUpCallback = wakeUpCallback;
 
     this.running = false;
   }
```

Comment:
Overloaded constructor so that this doesn't break anything that currently uses Daemon